docs/proposals/PROPOSAL_F.md

```diff
@@ -0,0 +1,182 @@
+
+## Image Compatibility Artifact
+
+In [Proposal B](PROPOSAL_B.md) it is stated that a compatibility artifact should describe CRITICAL requirements for container images to run on a host, which is inferred to mean a limited set that determines compatibility. This proposal takes the stance that we don't yet know what that minimal set is, and further, that it may vary depending on the use case for any specific artifact. Instead of declaring a minimum set, this proposal states that the artifact should contain minimum requirements for compatibility but can contain more that MAY be used by plugins to not just determine compatibility but optimal usage. The artifact serves as a source of metadata about container requirements, and it is on the level of the plugins or associated tooling to decide which subset should be used for a given use case. Arguably, this set will change. More discussion and research is required to understand what the line between critical (for the image to run, compatibility) and running optimally.
+
+### Artifact Manifest
+
+This artifact MAY have a manifest that references it with the compatibility artifact media type. For example:
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.oci.image-compatibility.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image-compatibility.spec.v1+json",
+      "digest": "sha256:4a47f8ae4c713906618413cb9795824d09eeadf948729e213a1ba11a1e31d052",
+      "size": 1710
+    }
+  ],
+  "subject": {
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270",
+    "size": 7682
+  }
+}
+```
+
+Note that the subject points to the original image that the artifact describes.
+
+### Artifact
+
+The artifact is a simple json structure under the top level key "compatibilities" of named groups and within them, key value pairs, where it is strongly suggested that each top level key refers to a compatibility group. Each compatibility special interest group that maintains an underlying namespace can choose the specific key value pairs within. Here is an example with NFD annotations. Note that the specific identifier for NFD is likely more complex than just "nfd" (e.g., io.kubernetes.nfd):
+
+```json
+{
+  "compatibilities": {
+    "nfd": {
+      "cpu-hardware_multithreading": true,
+      "cpu-pstate.status": "passive",
+      "cpu-pstate.turbo": false
+      "kernel-selinux.enabled": true,
+      "kernel-version.major": 4           
+      "memory-numa": true,
+      "memory-nv.dax": true
+      "network-sriov.capable": true,
+      "network-sriov.configured": true
```

Comment:
I have one question (it's common to all proposals, can be dicusssed later in label definition phase), if one compatibility label is introduced in new schema, and the tool(for exampl runtime) can't recognize the label because the tool itself can only support older schema. This unrecognized label should be handled comaptible or incompatible?

Reply:
I think the compatibility interest group maintaining the section would decide on a strategy for missing labels within the context of a tool. It could be considered a mismatch, ignored, or a version provided to scope to. I don’t think that’s something we can decode - it could vary by the use case. We could provide examples for different cases and options to the interest groups though.